cw_decred/lib/api/libdcrwallet.dart

```diff
@@ -0,0 +1,71 @@
+import 'dart:ffi';
+import 'dart:io';
+import 'package:flutter/foundation.dart';
+import 'package:cw_decred/api/libdcrwallet_bindings.dart';
+import 'package:cw_decred/api/util.dart';
+
+final String libraryName =
+    Platform.isAndroid || Platform.isLinux // TODO: Linux.
+        ? 'libdcrwallet.so'
+        : 'cw_decred.framework/cw_decred';
+
+final dcrwalletApi = libdcrwallet(DynamicLibrary.open(libraryName));
+
+/// initLibdcrwallet initializes libdcrwallet using the provided logDir and gets
+/// it ready for use. This must be done before attempting to create, load or use
+/// a wallet.
+void initLibdcrwallet(String logDir) {
+  final cLogDIr = logDir.toCString();
+  handleErrorAndPointers(
+    fn: () => dcrwalletApi.initialize(cLogDIr),
+    ptrsToFree: [cLogDIr],
+  );
+}
+
+class LoadWalletOptions {
+  final Pointer<Char> network;
+  final Pointer<Char> name;
+  final Pointer<Char> dataDir;
+
+  LoadWalletOptions({required String name, required String dataDir})
+      : network = "testnet".toCString(),
+        this.name = name.toCString(),
+        this.dataDir = dataDir.toCString();
+}
+
+class CreateWalletOptions extends LoadWalletOptions {
+  final Pointer<Char> password;
+
+  CreateWalletOptions(
+      {required super.name, required super.dataDir, required String password})
+      : this.password = password.toCString();
+}
+
+/// createWalletAsync calls the libdcrwallet's createWallet function
+/// asynchronously.
+Future<void> createWalletAsync(CreateWalletOptions opts) {
+  return compute(createWalletSync, opts);
```

Comment:
Turns out dart doesn't allow passing a class to compute. I'll change to use map as is done in `cw_monero`.